test/test_limits.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright 2025 Matt Borland
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+
+#include <boost/decimal.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+#if defined(__cpp_consteval) && __cpp_consteval >= 201811L && !defined(BOOST_DECIMAL_NO_CONSTEVAL_DETECTION)
+
+using namespace boost::decimal;
+
+template <typename T>
+consteval bool test_value(T value, const char* res)
```

Comment:
Would passing res as string_view here made logic a bit simpler? 
Then you would only need to make the string_view from (buff, r.ptr) and return if they are same?